### ISSUE TITLE
Can't compile sixel-sys on windows or macOS (works fine in linux)

### ISSUE THREAD

**Issue by mmulet** [AdnoC/sixel-sys#1]
On Windows: I get this error:
```
rror: failed to run custom build command for `sixel-sys v0.3.1`

Caused by:
  process didn't exit successfully: `V:\Documents\viuertest\sixel-sys-test\target\debug\build\sixel-sys-f5b3ec906222e763\build-script-build` (exit code: 101)
  --- stdout
  cargo:rustc-link-lib=dylib=sixel
  cargo:rustc-link-search=native=V:\Documents\viuertest\sixel-sys-test\target\debug\build\sixel-sys-060e93a697b4af5b\out\lib

  --- stderr
  thread 'main' panicked at 'Failed to execute ./configure: Os { code: 2, kind: NotFound, message: "The system cannot find the file specified." }', C:\Users\micha\.cargo\registry\src\github.com-1ecc6299db9ec823\sixel-sys-0.3.1\build.rs:71:22

```
On macOS I get this:
```
--- stderr
thread 'main' panicked at 'Failed to execute ./configure: OS {code: 2, kind: NotFound, message: "No such file or directory" }', /users/michaelmulet/.cargo/registry/src/github.com-1ecc6299db9ec823/sixel-sys-0.3.1/build.rs:71:22
configure: WARNING: unrecognized options: --without-python
configure : WARNING: unrecognized options: --without-python

```
If I try to build with RUST_BACKTRACE=1, then it says the error is at
```
build_script_build::main
at ./build.rs:71:9
```

Tracing it backwards, that just tells us that configure failed.

**Comment by mmulet** [comment]
Found the source of the error. It's a bug with std::proccess::Command.
tldr;
Create a dummy file named `configure` in the project root directory and everything works.
----
Actual Explanation:
In the current version of rust (1.49.0 at time of writing) the std::proccess::Command constructor `std::process::Command::new` takes in a &str as an argument. This argument is then interpreted in a OS dependent way. 
On macOS, but not Linux,  `std::proccess::Command::new` will attempt to find the command if it is a relative path. However, this is not the location of the actual command because we use the `.current_dir()` method to change the current working directory. 
So, on macOS, at `std::proccess::Command::new` the relative path you pass in must be a valid path relative to the working directory of the calling proccess **and** a valid path relative to the directory passed to `.current_dir()`.

I will create a bug report for the rust standard library, but in the meantime you can fix this with an ease workaround:
Create a dummy file named `configure` in the project root directory. This way the relative path will be valid in both contexts.

**Comment by mmulet** [comment]
For future reference here is the rust bug report: https://github.com/rust-lang/rust/issues/80819

**Comment by mmulet** [comment]
PR for fix:https://github.com/rust-lang/rust/pull/80537

**Comment by retpolanne** [comment]
Was able to build sixel-sys this way: 

```sh
touch libsixel/configure
chmod +x libsixel/configure
cargo build
```

**Comment by AdnoC** [comment]
I haven't touched this in a while, but I'll fix when I have the time.

**Comment by pjschinske** [comment]
I seem to be getting this bug on Archlinux and FreeBSD, even after creating the `configure` file as it is in mmulet's commit. Am I missing something?

**Comment by sravanpannala** [comment]
Do you have any updates on this issue?

**Comment by AdnoC** [comment]
Working on this now. Sorry for the lack of response. Pretty sure I was out of town when I replied last and forgot about this by the time I got back. I've been swamped with a new job.

**Comment by AdnoC** [comment]
I have released a new version that should fix the issues here (along with rust-lang fixing the stdlib bug). Going to see if my macbook still boots to verify that this fixes compilation on MacOS, but it now compiles fine on ubuntu. I don't believe this will ever compile on Windows, as the `libsixel` library does not support direct Windows compilation, [requiring MinGW](https://github.com/saitoha/libsixel/tree/master?tab=readme-ov-file#cross-compiling-with-mingw).

**Comment by AdnoC** [comment]
I have tested the new version on my macbook. It compiles without issue. Probably won't be able to confirm mingw compilation tonight, but will try to get it done soon.

**Comment by AdnoC** [comment]
An update on Windows compilation: I swapped to the `autotools` crate to compile `libsixel`, but that errors when I try to compile in MinGW.

I'll see if github actions offers mac boxes and if so I'll close this issue and open a windows only one linking this.

**Comment by evanjs** [comment]
Not sure what things might look like on the CI/GitHub build end, but...
I managed to get this library compiling on Windows

First, I made sure to compile `libsixel` in a MSYS/MINGW64 environment:
```
CC=x86_64-w64-mingw32-gcc cross_compile=yes ./configure --host=x86_64-w64-mingw32
make
make install
```

Then, I had to ensure this library (`sixel-sys`) builds on Windows
Correctness aside, the below does result in a functioning library on my system.

<details>
<summary><h2><span><code>build.rs</code> changes</span></h2></summary>

```diff
diff --git a/build.rs b/build.rs
index c604969..a1d3da8 100644
--- a/build.rs
+++ b/build.rs
@@ -45,7 +45,20 @@ fn main() {
         sixel_prefix("build")
     } else { sixel_build_dir.clone().into_os_string().into_string().expect("Could not convert OS path to utf8") };

-    {
+    if cfg!(windows) {
+        // Path to your Cygwin installation
+        let cygwin_prefix = Path::new("C:/msys64/mingw64/");
+
+        // Link to the prebuilt libsixel
+        println!("cargo:rustc-link-search=native={}/lib", cygwin_prefix.display());
+        println!("cargo:rustc-link-lib=sixel"); // or "sixel-static" if you have a static lib
+
+        // Optionally, tell bindgen where to find headers
+        println!("cargo:include={}/include", cygwin_prefix.display());
+
+        // Only rerun build.rs if the library changes
+        println!("cargo:rerun-if-changed={}/lib/libsixel.a", cygwin_prefix.display());
+    } else {
         let mut bld = Config::new("libsixel");
         if curl {
             bld.with("libcurl", None);
@@ -162,11 +175,7 @@ fn has_feature(feature: &'static str) -> bool {
 }

 fn sixel_prefix(directory: &str) -> String {
-    let cmd = Command::new("pwd").output().expect("Could not run `pwd`");
-    let base_path = std::str::from_utf8(&cmd.stdout)
-      .expect("Could not turn libsixel path into utf8")
-      .trim()
-      .to_string();
-    let path = base_path + "/" + LIBSIXEL_DIR + "/" + "build";
-    path
+    let cwd = env::current_dir().unwrap();
+    let path = cwd.join(LIBSIXEL_DIR).join("build");
+    path.display().to_string()
 }
```

</details>

## Notes

I am not sure if all the `println!` statements are required, most of this portion was vibe-coded tbh lol

I also had to update `sixel_prefix` with something cross-platform as `pwd` is not a builtin command on Windows
We don't actually need `Command`, let alone `pwd`, though, and can leverage `env::current_dir()` and `.join()` to achieve the same result
___
As @AdnoC suggested, though, we can carry this over to a new issue pertaining only to the Windows side if things extend beyond just the above, etc.
